lib/public/views/Runs/Runs.js

```diff
@@ -196,12 +196,37 @@ export default class Overview extends Observable {
 
     /**
      * Filter runs based on the selected run numbers and fields to prepare it for export
-     * @param {Object} runs The runs data
      * @returns {Object} The Filtered runs data
      */
-    async getFilteredRuns(runs) {
-        const selectedRuns = runs.filter((run) => this.getSelectedRunNumbers().includes(run.runNumber));
-        const filteredRuns = selectedRuns.map((selectedRun) => pick(selectedRun, this.getSelectedRunsFields()));
+    async getFilteredRuns() {
+        this.runs = RemoteData.loading();
+        this.notify();
+
+        const runIds = this.getSelectedRunNumbers().split(',');
```

Comment:
For `runIds` we know that we expect only valid numbers.
I would recommend to have an extra check in here that the values parsed are indeed numbers and no empty values are passed to the server.
If there is an empty value, it should be dropped from the list of Run Numbers.
If there are no valid numbers in the input, an error message should be printed.

e.g. If by mistake the user inputs the value: `2, 3, ,4,7`, this will generate an empty line in the CSV format